src/components/CommentCard.jsx

```diff
@@ -1,10 +1,55 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../contexts/User";
+import { deleteCommentById } from "../api";
 
-export default function CommentCard({ comment }) {
+export default function CommentCard({ comment, setComments }) {
   const parsedDate = Date.parse(comment.created_at);
+
   const formattedDate = new Date(parsedDate).toLocaleString("en-GB", {
     timeZone: "UTC",
   });
+
+  const { loggedInUser, setLoggedInUser } = useContext(UserContext);
+  const [deleteBtn, setDeleteBtn] = useState(false);
+  const [deleteComment, setDeleteComment] = useState({});
+
+  useEffect(() => {
+    loggedInUser.username === comment.author
+      ? setDeleteBtn(true)
+      : setDeleteBtn(false);
+  }, [loggedInUser]);
```

Comment:
This doesn't need to be in a useEffect - it would make more sense to do this conditional check as part of a [conditional render](https://legacy.reactjs.org/docs/conditional-rendering.html). Check out the linked page for a refresher on how that works!